src/main/kotlin/org/elm/ide/refactoring/extractExpressionUI.kt

```diff
@@ -0,0 +1,40 @@
+package org.elm.ide.refactoring
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.util.Pass
+import com.intellij.refactoring.IntroduceTargetChooser
+import org.elm.lang.core.psi.ElmExpressionTag
+import org.elm.openapiext.isUnitTestMode
+import org.jetbrains.annotations.TestOnly
+
+fun showExpressionChooser(
+        editor: Editor,
+        exprs: List<ElmExpressionTag>,
+        callback: (ElmExpressionTag) -> Unit
+) {
+    if (isUnitTestMode) {
+        callback(MOCK!!.chooseTarget(exprs))
+    } else {
+        IntroduceTargetChooser.showChooser(editor, exprs, callback.asPass) { it.text }
```

Comment:
nit: `asPass` is pretty gnarly for something only called once. It might be more readable to create the object manually:

```kotlin
val pass = object : Pass<ElmExpressionTag>() {
    override fun pass(t: ElmExpressionTag) = callback(t)
}
IntroduceTargetChooser.showChooser(editor, exprs, pass) { it.text }
```

Reply:
This util came from intellij-rust. Once the interface grows to include more hooks, it will make more sense to have this thing extracted out. So I'm going to leave it.